v2/workloadapi/option.go

```diff
@@ -6,25 +6,53 @@ import (
 	"google.golang.org/grpc"
 )
 
+type clientOption struct {
+	address     string
+	dialOptions []grpc.DialOption
+	log         logger.Logger
+}
+
 // ClientOption is an option used when creating a new Client.
-type ClientOption interface{}
+type ClientOption interface {
+	apply(*clientOption)
+}
+
+type funcClientOption struct {
+	f func(*clientOption)
+}
+
+func (fdo *funcClientOption) apply(do *clientOption) {
+	fdo.f(do)
+}
+
+func newFuncDialOption(f func(*clientOption)) *funcClientOption {
+	return &funcClientOption{
+		f: f,
+	}
+}
```

Comment:
This is a little simpler if you forgo the struct and just make a new type out of the function matching the right signature.

```suggestion
type funcClientOption func(*clientOption)

func (fn funcClientOption) apply(do *clientOption) {
	fn(do)
}
```
The functions below can just do:
```
return funcClientOption(func(c *clientOption) {
})
```